File: C#/day4.cs
class Person
{
    public int age;

    public Person(int initial_Age)
    {
        age = initial_Age < 0 ? 0 : initial_Age;
        if (initial_Age < 0)
            Console.WriteLine("This person is not valid, setting age to 0.");
    }
    public void amIOld()
    {
        if (age < 13)
            Console.WriteLine("You are young.");
        else if (age >= 13 && age < 18)
            Console.WriteLine("You are a teenager.");
        else
            Console.WriteLine("You are old.");
    }

    public void yearPasses()
    {
        age++;
    }
}

File: JAVASCRIPT/day4.js
function Person(initial_Age){
    // Add some more code to run some checks on initial_Age
    this.age = Number(initial_Age);
    if (!isNaN(this.age)) {
        if (this.age < 0) {
            this.age = 0;
            console.log('This person is not valid, setting age to 0.');
        }
    }

    this.amIOld=function(){
        // Do some computations in here and print out the correct statement to the console
        var out;

        if (this.age < 13) {
           out = 'You are young.';
        } else if (this.age >= 13 && this.age < 18) {
           out = 'You are a teenager.';
        } else {
           out = 'You are old.';
        }

        console.log(out);
    };

    this.yearPasses=function(){
          // Increment the age of the person in here
       this.age++;
    };
}